### PR TITLE
Adds log field for "reason" returned with status error codes

### PR DESCRIPTION
**Pull request by randall**
reason sometimes has more detail about why a request failed than the
status code or the result code (crc)

`prrr`: Proxy Response Result Reason

### CODE REVIEW

File: doc/admin-guide/logging/formatting.en.rst [hunk 1]
```diff
@@ -605,6 +606,7 @@ pssc  Proxy Response        HTTP response status code sent by |TS| proxy to the
                             client.
 sssc  Origin Response       HTTP response status code sent by the origin server
                             to the |TS| proxy.
+prrr  Proxy Server          Proxy response result reason
```
Comment (mlibbey) [inline doc/admin-guide/logging/formatting.en.rst]: Is this the Message that goes along with the Status Code? eg, the "OK" in 200 OK, and Not Modified in "304 Not Modified"?  If so, wonder if "message phrase" or "response phrase" would make it more clear? (both seem to be used in rfc 7231.

Comment (randall) [reply doc/admin-guide/logging/formatting.en.rst]: Good point. I've renamed it to "reason phrase" and updated things accordingly.

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Comment by zwoop** [comment]
Cherry-picked to 7.1.x

**Pull request merged**